db/memorystore.go

```diff
@@ -170,6 +172,13 @@ func (m *MemoryStore) AddOrder(order *core.Order) (int, error) {
 		return 0, fmt.Errorf("order %q already exists", orderID)
 	}
 
+	var ordersByAccountID []*core.Order
+	var present bool
+	if ordersByAccountID, present = m.ordersByAccountID[order.AccountID]; !present {
+		ordersByAccountID = make([]*core.Order, 0)
+	}
+	m.ordersByAccountID[order.AccountID] = append(ordersByAccountID, order)
```

Comment:
I think this is appending to the wrong slice:

```suggestion
	m.ordersByAccountID[order.AccountID] = append(m.ordersByAccountID, order)
```

Reply:
Either it should be `m.ordersByAccountID[order.AccountID]`, or `ordersByAccountID`. I used the latter because the former might not be defined yet.